Source/PLCrashLogWriter.m

```diff
@@ -577,6 +590,43 @@ void plcrash_log_writer_free (plcrash_log_writer_t *writer) {
     }
 }
 
+static long getFreeStorage()
+{
+    NSNumber *freeStorage = [[NSFileManager defaultManager]
```

Comment:
This (and similar places) breaks async-safety. Please read this doc - https://github.com/microsoft/plcrashreporter/blob/b1a342da19ed9b3af61ea2efa7656c2af30aeb7c/Source/CrashReporter.h#L213-L239

Reply:
I  updated this function to not use an objective-c api to check the free disk space.

Reply:
Agreed. AFAIK there is no way to retrieve the battery level without using UIDevice (correct me if I'm wrong), which is objc runtime as you noted. So we can't really use such code at the time of writing the crash info.

As for the `getFreeStorage`, for this you can use `customData` API:
https://github.com/microsoft/plcrashreporter/blob/b1a342da19ed9b3af61ea2efa7656c2af30aeb7c/Source/PLCrashReporter.h#L153
Working with file system can be tricky, I would avoid doing that at the crash time.

For the memory info, as a workaround you can store in `customData` as well, or use a callback
https://github.com/microsoft/plcrashreporter/blob/b1a342da19ed9b3af61ea2efa7656c2af30aeb7c/Source/PLCrashReporter.h#L143